File: src/test/java/com/kobylynskyi/graphql/codegen/scala/compile-pass/EventsCreatedSubscriptionResponse.scala
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResult
import java.util.{ Map => JMap }

/**
 * Subscribe to events
 */
class EventsCreatedSubscriptionResponse extends GraphQLResult[JMap[String, scala.Seq[Event]]] {

    /**
     * Subscribe to events
     */
    def eventsCreated(): scala.Seq[Event] = {
        val data: JMap[String, scala.Seq[Event]] = getData
        if (data != null) data.get(EventsCreatedSubscriptionResponse.OPERATION_NAME) else null
    }

}

object EventsCreatedSubscriptionResponse {

    private final val OPERATION_NAME: String = "eventsCreated"

}

File: src/test/java/com/kobylynskyi/graphql/codegen/kotlin/compile-pass/EventsCreatedSubscriptionResponse.kt
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResult

/**
 * Subscribe to events
 */
open class EventsCreatedSubscriptionResponse : GraphQLResult<MutableMap<String, List<Event>>>() {

    companion object {
        const val OPERATION_NAME: String = "eventsCreated"
    }

    /**
     * Subscribe to events
     */
    fun eventsCreated(): List<Event> {
        val data: MutableMap<String, List<Event>> = super.getData()
        return data.getValue(OPERATION_NAME)
    }
}
